### ISSUE TITLE
testnet stuck

### ISSUE THREAD

**Issue by l-n-s** [majestrate/XD#43]
## info

git revision / version: master

OS: linux

Architecture: x86

## problem

Doesn't seem to download anything

## backtrace / error messages

Error messages: [yes/no]

seeder

```
[DBG] 2018-01-30 16:04:58.988580001 +0000 UTC m=+27.148727689   torrent obtained network
[DBG] 2018-01-30 16:05:58.990329845 +0000 UTC m=+87.150477557   http://uhjd472zbgcipmkv4yfn7seo6m7gz3vmqqc5lhj2tsn6erkfn2fq.b32.i2p/a announcing
[NFO] 2018-01-30 16:05:59.231566508 +0000 UTC m=+87.391714166   http://uhjd472zbgcipmkv4yfn7seo6m7gz3vmqqc5lhj2tsn6erkfn2fq.b32.i2p:80/a got 1 peers for 87f1107326c2260b26183bb931b1a7d8185fce18
[DBG] 2018-01-30 16:05:59.245131258 +0000 UTC m=+87.405278913   wrote 68 of 68

```

leecher

```
[DBG] 2018-01-30 16:04:58.897221855 +0000 UTC m=+26.113240379   torrent obtained network
[DBG] 2018-01-30 16:05:58.898830484 +0000 UTC m=+86.114848961   http://uhjd472zbgcipmkv4yfn7seo6m7gz3vmqqc5lhj2tsn6erkfn2fq.b32.i2p/a announcing
[NFO] 2018-01-30 16:05:59.00738353 +0000 UTC m=+86.223402222    http://uhjd472zbgcipmkv4yfn7seo6m7gz3vmqqc5lhj2tsn6erkfn2fq.b32.i2p:80/a got 0 peers for 87f1107326c2260b26183bb931b1a7d8185fce18
[DBG] 2018-01-30 16:05:59.334553519 +0000 UTC m=+86.550572048   got inbound bittorrent connection from 0D2lt-g~8BXG4vXgWPw0OBqz7yR5BVxsucF1eWDB~8qIKvXpfSmMhqWVPBkPxtzDeyKBcbCt~kVNJQVLGpJ3keJIAWlDCMYERRkBp4WkGqTysqElRaEhxtakhyofa294uOXKva2U6m0y1893TaQaHXBYpph83FOWD~kBF9dcFmNa3PykrTeyrUcikMUht18GD8w72W14cGm8jz6HRyl51Zh-8m-cF8W~gnchSDt709sVXHc94~KvcuP1MK6K3tqa5DWeX3oA74lvQsbawOKz1JVlnjLabRluv4RxMF646qiQXFXkBwZG-~4Ogn~WV87TMsksMm5EL1O~QB4Vx2-MAG-N9iQttrBjrLcph6tlpRL2HTA6oGBqNMCKYADxvpz4SdMnYWwwf5fI-f5g6DDdyLHBi2XGGol~7wi-qw5TBMuf0dolk~FDCaCjXbB24lSjGiqj8vHBHO5j2XHtijwAmXOoM4FbIbz9R0WEu-rYXqmRLBviKeY1XU4yuF483ksFAAAA
```

Backtrace: [yes/no]

[insert any backtraces here]

**Comment by l-n-s** [comment]
wow it finished finally

```
XD instances are ready, waiting for download to finish...
Leecher file is downloaded
It took only 2177.3034603595734 seconds
Shutting down, please wait...
```

**Comment by l-n-s** [comment]
same happens again

Seeder:

```
[DBG] 2018-02-01 11:37:25.379451135 +0000 UTC m=+21.106878664   sum len=262144 idx=780 ih=800
[DBG] 2018-02-01 11:37:25.380203339 +0000 UTC m=+21.107630886   piece 40 of 40
[DBG] 2018-02-01 11:37:25.381658448 +0000 UTC m=+21.109086001   torrent obtained network
[DBG] 2018-02-01 11:38:25.385151259 +0000 UTC m=+81.112578802   http://5rcm7iqg7tyroadipfu7r4flff26dwbuv4e5inntlraojsru3rcq.b32.i2p/a announcing
[NFO] 2018-02-01 11:38:25.466453128 +0000 UTC m=+81.193880746   http://5rcm7iqg7tyroadipfu7r4flff26dwbuv4e5inntlraojsru3rcq.b32.i2p:80/a got 0 peers for d02f38fc97595025f8a2fbf418615e18a3fcda97
[DBG] 2018-02-01 11:38:26.029635361 +0000 UTC m=+81.757062927   got inbound bittorrent connection from 2am5zToRUhOfz9s2wlBWMjTd~7wzDL2EZh-TmDM7WC9wnXuVYK-ZlxMK3Yt3IofL56zyIIbwsMrjEMn-GaV~ij0MaWSyFDJ1FCvxXk62NAsxQzdjq2WSoSMjwTUSlr1C8YfjLExvGNlQB8Hmh8g7~~Mv3xIRkrsnHCmCXs~03Qcm7YWPcoxMV~2eWszvZUvn9CBcjK~YExg811L13jPto~swzKnLNb7b-y061ivC3QGsLP9cDCK-a~djn-k1ndJSsz1AaTrl6eLi5V96wfrE85F1k-9dyufkv5IPwBSttKodipCRHvyeYrlQGl2oqlQxpj~hdwRIQ1TN2OgZxVt8EIhPfUfF5DcJKHK-gJZvCkMTY8J-cZw15L2yOi98VLYiFS8KzdeV3-XhvHcForf4Yi0XQOJYQGm80XGxUED4XSii459gpjXwmVpHlB4iW0lWKs7BRd9QJpRiAm6raqdBRxN8f0sCAgUVPidtVV3hUeMBKuUCc-BW~RXtF3GIAbDxAAAA
```

leecher:

```
[DBG] 2018-02-01 11:37:25.837635574 +0000 UTC m=+21.083528392   piece 40 of 40
[DBG] 2018-02-01 11:37:25.838455638 +0000 UTC m=+21.084348457   torrent obtained network
[DBG] 2018-02-01 11:38:25.83974112 +0000 UTC m=+81.085633896    http://5rcm7iqg7tyroadipfu7r4flff26dwbuv4e5inntlraojsru3rcq.b32.i2p/a announcing
[NFO] 2018-02-01 11:38:25.946498087 +0000 UTC m=+81.192390905   http://5rcm7iqg7tyroadipfu7r4flff26dwbuv4e5inntlraojsru3rcq.b32.i2p:80/a got 1 peers for d02f38fc97595025f8a2fbf418615e18a3fcda97
[DBG] 2018-02-01 11:38:25.978432944 +0000 UTC m=+81.224325755   wrote 68 of 68
```

**Comment by l-n-s** [comment]
Seems to start downloading after 30 or 35 minutes... may be related with https://github.com/PurpleI2P/i2pd/issues/1072

**Comment by majestrate** [comment]
How many passive nodes and how many active nodes in test net?

**Comment by l-n-s** [comment]
7 floodfills and 6 nodes. 2 XD nodes and 1 opentracker.

Now XD tests share i2pd testnet state, and nodes are picked randomly for each run, so periodically they all have activity. 

This bug is rare, there is a hack now, tests notice if XD is stuck.

**Comment by majestrate** [comment]
XD idles until the SAM session is ready, it's probably because no outbound tunnels are being built fast enough.
